jaxrs-api/src/main/java/jakarta/ws/rs/Path.java

```diff
@@ -65,6 +68,8 @@
 @Target({ ElementType.TYPE, ElementType.METHOD })
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
+@Stereotype
+@RequestScoped
```

Comment:
Can it possibly clash with @Singleton?

Reply:
TBH I don't know how that works since `@Singleton` is not part of CDI. However, resources should be `@RequestScoped` per the specification documentation. I don't see anything in the CDI specification that clears this up.

Reply:
CDI says `@Singleton` is the [pseudoscope](https://docs.jboss.org/weld/reference/latest/en-US/html/scopescontexts.html#_the_singleton_pseudo_scope). Which I am not sure how it is handled by Weld. Would need to do some testing.

However, the REST spec supports the `@jakarta.inject.Singleton` resource classes, for instance in [Section 9.4](https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1#sse_broadcasting).

It also [mandates](https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1#ejbs):
> In a product that supports EJBs, an implementation MUST support the use of stateless and singleton session beans as root resource classes

Though it is jakarta.ejb.Singleton, I still feel some potential clash with `@RequestScoped`.

Reply:
This is what Weld throws:

Reply:
That only happens if you add both `@RequestScoped` and `@Singleton` to the resource though. If you only add `@Singleton` and `@Path` it will work. A stereotypes scope can be overridden as it's just the default scope. From the end of https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0#stereotype_default_scope

> For example, the following stereotype might be used to identify action classes in a web application:
>
> ```
> @RequestScoped
> @Stereotype
> @Target(TYPE)
> @Retention(RUNTIME)
> public @interface Action {}
> ```
> Then actions would have scope @RequestScoped unless the scope is explicitly specified by the bean.

Reply:
Thank you, I was not aware of it, that resolves my concerns. Keeping unresolved for others should they have the same doubts.